scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java

```diff
@@ -66,31 +66,30 @@ private void getJobXML(HttpServletRequest request, HttpServletResponse response)
         String jobId = request.getParameter("jobId");
         String sessionId = request.getParameter("sessionId");
         String fileName = "job_" + jobId + ".xml";
-        String location = "job" + jobId + ".result";
 
-        ServletOutputStream out = null;
-        try (InputStream is = new ByteArrayInputStream(((SchedulerServiceImpl) Service.get()).getJobXML(sessionId,
-                                                                                                        jobId)
-                                                                                             .getBytes(StandardCharsets.UTF_8))) {
+        try (ServletOutputStream out = response.getOutputStream();
+                InputStream is = new ByteArrayInputStream(((SchedulerServiceImpl) Service.get()).getJobXML(sessionId,
+                                                                                                           jobId)
+                                                                                                .getBytes(StandardCharsets.UTF_8))) {
             response.setContentType("application/xml");
             response.setHeader("Content-disposition", "attachment; filename=" + fileName);
-            response.setHeader("Location", location);
-
-            out = response.getOutputStream();
+            response.setHeader("Location", fileName);
 
             int buffer;
             while ((buffer = is.read()) != -1) {
                 out.write(buffer);
+                out.flush();
```

Comment:
I think `flush` is not needed here. Because it will flush every byte, isn't it?
Try with resources will auto close your streams. Which will execute flush as well

Reply:
And plus I don't see why it would be needed to flush at each iteration.

Reply:
@jrochas It should be out of the loop, I don't know how I put it inside
@Mykhailenko in Closeables are not flushed, only closed. No where in the java doc it's mentioned, except for `Writer`.